### DISCUSSION TITLE
Plz add docs keepPreviousData for useContractReads

### DISCUSSION THREAD

**Discussion by nasa8x** [wevm/wagmi#697] [General]
It took me a long time to figure out why every new block of `data` returned `undefined` before the request was completed. Until I had to read the source code, I discovered the `keepPreviousData` option and simply setting the value to `true` fixed the problem.

**Comment by jxom** [comment]
What version of wagmi are you using? The behaviour by default should be to keep the previous data on each block without setting the `keepPreviousData` flag.

**Reply by nasa8x** [reply]
I use version `^0.5.9`. I also think it makes sense to leave the default as `true`. <https://github.com/tmm/wagmi/blob/main/packages/react/src/hooks/contracts/useContractReads.ts#L67>
